### PR TITLE
Using Jimp instead of Paper.js

### PR DESCRIPTION
**Pull request by Lorti**
These changes address #1. 

I replaced `asset-resolver`, `image-size` and `paper` with a single dependency to `jimp`. Wasn't able to get 100% test coverage though… any ideas on that?

### CODE REVIEW

**Comment by ben-eb** [comment]
Hi @Lorti,

Thanks! This looks really good. I think we can address the code coverage by trying to load an unsupported image type; we could go with an obsoleted image format, or we could try to load some other binary format that Jimp doesn't support. Some small fixture in `/src/__tests__` would be good. What do you think?

**Comment by Lorti** [comment]
Thanks, you were right about the format Jimp doesn't support. Now waiting for Travis to build 🙂

**Comment by ben-eb** [comment]
Seems like Node 4 on OS X is still not playing ball. Perhaps we could simplify the Travis config further?

``` yml
sudo: false

language: node_js
matrix:
  include:
    - node_js: '6'
      os: linux
    - node_js: '5'
      os: linux
    - node_js: '4'
      os: linux
    - node_js: '6'
      os: osx
    - node_js: '5'
      os: osx
    - node_js: '4'
      os: osx
```

We don't need either `sudo` or any extra dependencies as we removed Cairo.

**Comment by Lorti** [comment]
I haven't had any experience with Travis prior to today, but it seems that it's been choking on the Node 4 build even before my changes – taking as long as the other five builds combined. Not sure what do to right now.

**Comment by ben-eb** [comment]
Yeah, it's been a bit problematic for me also, but I put this down to the extra hoops that I had to jump through to get Cairo running on CI. Could you try with the exact config that I posted in the last comment please? If we can't get it to pass I'll try contacting support and see if they know what's going on.

File: .travis.yml [hunk 1]
```diff
@@ -1,42 +1,16 @@
...
-      node_js: '6'
+    - node_js: '6'
       os: osx
-    - env: CXX=g++-4.9
-      node_js: '5'
+    - node_js: '5'
       os: osx
-    - env: CXX=g++-4.9
-      node_js: '4'
+    - node_js: '4'
```
Comment (ben-eb) [inline .travis.yml]: Still problematic. I think we should remove the Node 4 tests for OS X, they take too long on Travis.

File: src/__tests__/index.js [hunk 2]
```diff
@@ -160,3 +161,15 @@ test(
...
+test(
+    'should error on non-existing image',
+    shouldThrow,
+    `header{background:resemble-image(url(), twenty-five)}`
+);
+
+test(
+    'should error on unprocessable image',
+    shouldThrow,
+    `header{background:resemble-image(url("${unprocessable}"), twenty-five)}`
```
Comment (ben-eb) [inline src/__tests__/index.js]: I think I would prefer that we tested this with an empty value for the fidelity, as the module also throws when the fidelity is incorrect. That way we can be sure that we are testing the invalid file and not an invalid fidelity.

File: src/resembleImage.js [hunk 3]
```diff
@@ -32,36 +30,41 @@ function resolveFidelity (width, pair) {
...
-                        height
-                    );
-                    stops.push(colourStop(raster.getAverageColor(rect).toCSS(true), i));
-                }
+function rgbToHex ({r, g, b}) {
+    function convert (c) {
+        const hex = c.toString(16);
+        return hex.length === 1 ? '0' + hex : hex;
+    }
+    return convert(r) + convert(g) + convert(b);
```
Comment (ben-eb) [inline src/resembleImage.js]: We need a leading `#` for this value; I used to have actual hex values in the tests but the values don't actually align across platforms. But we can update the tests to make sure that each stop is a hex colour of some sort:

``` js
function assertColourStops (t, fixture, expected, options) {
    return postcss(plugin(options)).process(fixture).then((result) => {
        valueParser(result.root.first.nodes[0].value).walk(node => {
            if (node.value !== 'linear-gradient') {
                return false;
            }
            const stops = getArguments(node).slice(1);
            t.deepEqual(stops.length, expected);
            stops.forEach(stop => {
                const colour = stop[0].value;
                t.truthy(
                    colour.length === 7 && !isNaN(parseInt(colour.slice(1), 16))
                );
            });
        });
    });
}
```

**Review by ben-eb — CHANGES_REQUESTED** [review]
Looks great so far! Just a few minor tweaks. 😄

**Comment by Lorti** [comment]
I have opted to use the regular expression `/^#[0-9a-f]{6}$/i` instead of `colour.length === 7 && !isNaN(parseInt(colour.slice(1), 16))`. This way the test also looks for the '#' sign.

**Review by ben-eb — APPROVED** [review]

**Comment by ben-eb** [comment]
Alright, thanks! 👍

**Pull request merged**